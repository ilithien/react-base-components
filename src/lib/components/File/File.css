.react-base__file {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.react-base__file > .file-label {
    background-color: #e74c3c;
    color: #eee;
    padding: 8px 15px 8px 15px;
    border-radius: 5px;    
}

.react-base__file > img {
    max-width: 250px;
}