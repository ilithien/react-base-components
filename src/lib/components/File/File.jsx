import React, { Component } from "react";
import PropTypes from "prop-types";
import "./File.css";

export default class FileInput extends Component {
  constructor(props) {
    super(props);
    this.state = {
      file: ""
    };

    this.onLoad = this.onLoad.bind(this);
  }

  onLoad({ target }) {
    const {
      files: [first, ...rest]
    } = target;

    if (first) {
      const reader = new FileReader();
      reader.onload = e => {
        this.setState({
          file: e.target.result
        }),
          () => {
            this.props.onLoadImage(e.target.result);
          };
      };
      reader.readAsDataURL(first);
    }
  }

  render() {
    const { id = "react-file-input", label = `Choose a file...` } = this.props;
    return (
      <div className="react-base__file">
        <img src={this.state.file} />
        <label className="file-label" htmlFor={id}>
          {label}
        </label>
        <input type="file" id={id} hidden onChange={this.onLoad} />
      </div>
    );
  }
}

FileInput.propTypes = {
  label: PropTypes.string,
  id: PropTypes.string,
  onLoadFile: PropTypes.func.isRequired
};
