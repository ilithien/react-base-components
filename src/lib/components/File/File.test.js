import React from "react";
import renderer from "react-test-renderer";
import File from "./File";

describe("Testing File input component", () => {
  it("matches snapshot", () => {
    const snap = renderer.create(
      <File onLoadImage={result => console.log(`Loaded image ${result}`)} />
    );

    const tree = snap.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
