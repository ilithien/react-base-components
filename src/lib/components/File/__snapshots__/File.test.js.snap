// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing File input component matches snapshot 1`] = `
<div
  className="react-base__file"
>
  <img
    src=""
  />
  <label
    className="file-label"
    htmlFor="react-file-input"
  >
    Choose a file...
  </label>
  <input
    hidden={true}
    id="react-file-input"
    onChange={[Function]}
    type="file"
  />
</div>
`;
